src/metadata/MetadataStorage.ts

```diff
@@ -102,7 +115,13 @@ export class MetadataStorage {
     });
 
     // get metadatas for inherited classes
-    const inheritedMetadatas = this.validationMetadatas.filter(metadata => {
+    const filteredForInheritedMetadatasSearch = [];
+    for (const [key, value] of this.validationMetadatas.entries()) {
+      if (targetConstructor.prototype instanceof key) {
+        filteredForInheritedMetadatasSearch.push(...value);
+      }
+    }
```

Comment:
Isn't it faster to loop over the prototype chain and look up the contructors in the map?

Reply:
I think you are right, but this I did not fix in the new PR. I just merged it in its current state for the performance gains. If you are up for the task, please implement it in a new PR.